test/Bart/GitHook/GitHookControllerTest.php

```diff
@@ -146,21 +170,37 @@ private function runProcessRevisionTest(array $stdInArray, array $revList, array
 
 		// The number of runs for $gitHookConfig->getValidRefs() depend on the total number of
 		// inputs in the standard input array and the number of revisions
-		$stubConfig = $this->shmock('\Bart\GitHook\GitHookConfig', function($gitHookConfig) use($numInputs, $validRefs) {
+		$stubConfig = $this->shmock('\Bart\GitHook\GitHookConfig', function($gitHookConfig) use($numInputs, $validRefs, $emergency) {
 			$gitHookConfig->getValidRefs()->times($numInputs)->return_value($validRefs);
+            if ($emergency){
+                if (!empty($validRefs)) {
+                    $gitHookConfig->getEmergencyNotificationBody()->return_value('');
+                    $gitHookConfig->getEmergencyNotificationBody()->return_value('');
+                    $gitHookConfig->getEmergencyNotificationEmail()->return_value('');
+                    $gitHookConfig->getEmergencyNotificationSubject()->return_value('');
+                }
+            }
+
 		}, true);
 
+        GlobalFunctions::register('mail', function($to, $subject, $body) {});
```

Comment:
It would be probably be useful to validate that the `to`, `subject`, and `body` values passed indeed the values we expected from the shmocked object.